src/routing.rs

```diff
@@ -1,30 +1,58 @@
-use axum::extract::Path;
-use axum::Router;
-use axum::routing::get;
+use std::sync::{Arc, RwLock};
+use axum::{
+    extract::{Path, State},
+    http::StatusCode,
+    routing::{get, post},
+    response::Response,
+    Router
+};
 use rust_intro::rust_intro::DataMessage;
+use crate::data::DataHandler;
 
-pub(crate) fn init_routes() -> Router {
+pub(crate) async fn init_routes() -> Router {
+    let handler = Arc::new(RwLock::new(DataHandler::new()));
     Router::new()
         .route("/", get(route_root))
         .route("/test", get(route_test))
         .route("/test/:path_param", get(route_test_with_param))
         .route("/proto/serialize/:path_param", get(route_serialize_testparam))
+        .route("/proto/:data_message_id/:message", post(route_add_data_msg))
+        .route("/proto/:data_message_id", get(route_get_data_msg))
+        .with_state(handler)
 }
 
-async fn route_root() -> &'static str {
+async fn route_root(_: State<Arc<RwLock<DataHandler>>>) -> &'static str {
     "Welcome to rust-intro"
 }
 
-async fn route_test() -> &'static str {
+async fn route_test(_: State<Arc<RwLock<DataHandler>>>) -> &'static str {
     "Test route"
 }
 
-async fn route_test_with_param(Path(path_param): Path<String>) -> String {
+async fn route_test_with_param(_: State<Arc<RwLock<DataHandler>>>, Path(path_param): Path<String>) -> String {
     format!("Test route with param: {}", path_param)
 }
 
-async fn route_serialize_testparam(Path(testparam): Path<String>) -> String {
-    let data = DataMessage {id: String::from("1"), content: testparam};
+async fn route_serialize_testparam(_: State<Arc<RwLock<DataHandler>>>, Path(testparam): Path<String>) -> String {
+    let data = DataMessage { id: String::from("1"), content: testparam };
 
     format!("{:?}", data)
 }
+
+async fn route_add_data_msg(State(state): State<Arc<RwLock<DataHandler>>>, Path((id, content)): Path<(String, String)>) -> StatusCode {
+    let mut lock = state.write().expect("could not get write lock");
```

Comment:
With the tokio RwLock, this must be changed.
```suggestion
    let mut lock = state.write().await;
```